### ISSUE TITLE
backend query task died unexpectedly: cannot commit transaction - SQL statements in progress

### ISSUE THREAD

**Issue by LefterisJP** [rotki/rotki#5381]
## Problem Definition

I noticed this only once in a big account, in production (1.26.3). Never encountered it before.

Got a message in the UI:
`Failed to fetch prices: The backend query task died unexpectedly: cannot commit transaction - SQL statements in progress`

## Logs

Checking the logs I saw this:

```
[11/01/2023 09:21:39 CET] DEBUG rotkehlchen.externalapis.defillama Greenlet-16: Querying defillama: https://coins.llama.fi/prices/current/ethereum:0x6B3595068778DD592e39A122f4f5a5cF09C90fE2?
[11/01/2023 09:21:39 CET] DEBUG rotkehlchen.inquirer Greenlet-5: Current price oracle defillama got price from_asset=eip155:1/erc20:0x7D1AfA7B718fb893dB30A3aBc0Cfc608AaCfeBB0(Matic Network), to_asset=USD(United States Dollar), price=0.860085
[11/01/2023 09:21:39 CET] ERROR rotkehlchen.api.rest Greenlet with id 140380916412256: Greenlet for task 20 dies with exception: cannot commit transaction - SQL statements in progress.
Exception Name: <class 'sqlite3.OperationalError'>
Exception Info: cannot commit transaction - SQL statements in progress
Traceback:
   File "src/gevent/greenlet.py", line 908, in gevent._gevent_cgreenlet.Greenlet.run
  File "rotkehlchen/api/rest.py", line 298, in _do_query_async
  File "rotkehlchen/api/rest.py", line 3386, in _get_current_assets_price
  File "rotkehlchen/inquirer.py", line 587, in find_price_and_oracle
  File "rotkehlchen/inquirer.py", line 534, in _find_price
  File "rotkehlchen/inquirer.py", line 624, in find_usd_price_and_oracle
  File "rotkehlchen/inquirer.py", line 657, in _find_usd_price
  File "rotkehlchen/assets/asset.py", line 168, in resolve_to_fiat_asset
  File "rotkehlchen/assets/resolver.py", line 155, in resolve_asset_to_class
  File "rotkehlchen/globaldb/handler.py", line 1871, in resolve_asset_from_packaged_and_store
  File "rotkehlchen/globaldb/handler.py", line 985, in edit_evm_token
  File "contextlib.py", line 126, in __exit__
  File "rotkehlchen/db/drivers/gevent.py", line 313, in write_ctx

[11/01/2023 09:21:39 CET] DEBUG rotkehlchen.externalapis.defillama Greenlet-15: Querying defillama: https://coins.llama.fi/prices/current/ethereum:0x6B3595068778DD592e39A122f4f5a5cF09C90fE2?
[11/01/2023 09:21:39 CET] DEBUG rotkehlchen.inquirer Greenlet-7: Current price oracle defillama got price from_asset=eip155:1/erc20:0x6DEA81C8171D0bA574754EF6F8b412F2Ed88c54D(LQTY), to_asset=USD(United States Dollar), price=0.5883
[11/01/2023 09:21:39 CET] DEBUG rotkehlchen.externalapis.defillama Greenlet-7: Querying defillama: https://coins.llama.fi/prices/current/ethereum:0xba100000625a3754423978a60c9317c58a424e3D?
```

### System Description

Operating system: Linux
Rotki version: 1.26.3

## Task

Figure out why it happens and fix it

**Comment by LefterisJP** [comment]
Have not seen this again so will close the issue

**Comment by LefterisJP** [comment]
I saw this again. In 1.27.1. So this is still possible. Happens very rarely but it can happen. No idea why, we would need to figure out the cause. Below is some logs excrepts.

```
...
[09/03/2023 00:34:32 CET] DEBUG rotkehlchen.api.server Greenlet-0: start rotki api POST /api/1/assets/prices/latest view_args={}, query_string=b''
[09/03/2023 00:34:32 CET] DEBUG rotkehlchen.api.server Greenlet-0: end rotki api POST /api/1/assets/prices/latest view_args={}, query_string=b'', status_code=200, result={'result': {'task_id': 25}, 'message': ''}
[09/03/2023 00:34:32 CET] DEBUG rotkehlchen.api.rest Greenlet-9: Async task with task id 25 started
...
[09/03/2023 00:34:33 CET] WARNING rotkehlchen.externalapis.defillama Greenlet-9: Queried Defillama current price from eip155:1/erc20:0x1985365e9f78359a9B6AD760e32412f4a445E862 to USD. But coins is not available in the result {'coins': {}}
[09/03/2023 00:34:33 CET] DEBUG rotkehlchen.externalapis.coingecko Greenlet-9: Querying coingecko: https://api.coingecko.com/api/v3/simple/price/?ids=augur&vs_currencies=usd
[09/03/2023 00:34:33 CET] DEBUG rotkehlchen.inquirer Greenlet-9: Current price oracle coingecko got price from_asset=eip155:1/erc20:0x1985365e9f78359a9B6AD760e32412f4a445E862(Augur), to_asset=USD(United States Dollar), price=7.56
[09/03/2023 00:34:33 CET] DEBUG rotkehlchen.externalapis.defillama Greenlet-9: Querying defillama: https://coins.llama.fi/prices/current/coingecko:tezos?
[09/03/2023 00:34:33 CET] DEBUG rotkehlchen.inquirer Greenlet-9: Current price oracle defillama got price from_asset=XTZ(Tezos), to_asset=USD(United States Dollar), price=1.032
[09/03/2023 00:34:33 CET] ERROR rotkehlchen.api.rest Greenlet with id 139997798872960: Greenlet for task 25 dies with exception: cannot commit transaction - SQL statements in progress.
Exception Name: <class 'sqlite3.OperationalError'>
Exception Info: cannot commit transaction - SQL statements in progress
Traceback:
   File "src/gevent/greenlet.py", line 908, in gevent._gevent_cgreenlet.Greenlet.run
  File "rotkehlchen/api/rest.py", line 312, in _do_query_async
  File "rotkehlchen/api/rest.py", line 3674, in _get_current_assets_price
  File "rotkehlchen/inquirer.py", line 589, in find_price_and_oracle
  File "rotkehlchen/inquirer.py", line 536, in _find_price
  File "rotkehlchen/inquirer.py", line 626, in find_usd_price_and_oracle
  File "rotkehlchen/inquirer.py", line 659, in _find_usd_price
  File "rotkehlchen/assets/asset.py", line 152, in resolve_to_fiat_asset
  File "rotkehlchen/assets/resolver.py", line 136, in resolve_asset_to_class
  File "rotkehlchen/globaldb/handler.py", line 1812, in resolve_asset_from_packaged_and_store
  File "rotkehlchen/globaldb/handler.py", line 986, in edit_evm_token
  File "contextlib.py", line 126, in __exit__
  File "rotkehlchen/db/drivers/gevent.py", line 372, in write_ctx

[09/03/2023 00:34:33 CET] ERROR rotkehlchen.api.rest Greenlet with id 139997802130176: Greenlet for task 5 dies with exception: cannot start a transaction within a transaction.
Exception Name: <class 'sqlite3.OperationalError'>
Exception Info: cannot start a transaction within a transaction
Traceback:
   File "src/gevent/greenlet.py", line 908, in gevent._gevent_cgreenlet.Greenlet.run
  File "rotkehlchen/api/rest.py", line 312, in _do_query_async
  File "rotkehlchen/api/rest.py", line 456, in _query_all_balances
  File "rotkehlchen/rotkehlchen.py", line 829, in query_balances
  File "rotkehlchen/utils/mixins/lockable.py", line 46, in wrapper
  File "rotkehlchen/utils/mixins/cacheable.py", line 122, in wrapper
  File "rotkehlchen/exchanges/binance.py", line 811, in query_balances
  File "rotkehlchen/exchanges/binance.py", line 213, in first_connection
  File "rotkehlchen/exchanges/utils.py", line 115, in query_binance_exchange_pairs
  File "rotkehlchen/globaldb/binance.py", line 40, in save_all_binance_pairs
  File "rotkehlchen/globaldb/handler.py", line 246, in add_setting_value
  File "contextlib.py", line 119, in __enter__
  File "rotkehlchen/db/drivers/gevent.py", line 360, in write_ctx
  File "rotkehlchen/db/drivers/gevent.py", line 89, in execute
```

**Comment by LefterisJP** [comment]
During testing for 1.28.0 I have not seen it. If we don't see it a bit more then I will probably close again.

**Comment by LefterisJP** [comment]
I am closing this as it has not been seen in some time. May reopen if we see it again (or make new issue if we forget this exists)

**Comment by mehdisadeghi** [comment]
I just wanted to open my rotki after a few months, and I got this error on Arch Linux. I was on rotki 1.39.1-1 from aur, I upgraded to 1.30.2-1 and still the same error. In the log file printed in terminal I see a relevant line:

    ValueError: Failed at global DB upgrade from version 3 to 4: no such table: address_book

I guess my database is corrupted.

**Comment by yabirgb** [comment]
> I just wanted to open my rotki after a few months, and I got this error on Arch Linux. I was on rotki 1.39.1-1 from aur, I upgraded to 1.30.2-1 and still the same error. In the log file printed in terminal I see a relevant line:
> 
> ```
> ValueError: Failed at global DB upgrade from version 3 to 4: no such table: address_book
> ```
> 
> I guess my database is corrupted.

Hello! From what exact release are you upgrading? 1.39.1 is ahead of our current release. Also we had this issue before but should be already fixed https://github.com/rotki/rotki/issues/6236. Could you confirm that you are using 1.30.2?

**Comment by mehdisadeghi** [comment]
> Could you confirm that you are using 1.30.2?

Yes. According to rotki About page:

App Version: 1.30.2
Frontend Version: Frontend Version:
Platform: linux x64 6.5.4-arch2-1
Electron: 25.8.1

I think there was problems doing multiple migrations after a long time and it broke. I can't say for sure that it is rotki's fault, perhaps one migration is ended abrupty (it was not responding for a long time and I killed the terminal).

**Comment by LefterisJP** [comment]
> > Could you confirm that you are using 1.30.2?
> 
> Yes. According to rotki About page:
> 
> App Version: 1.30.2 Frontend Version: Frontend Version: Platform: linux x64 6.5.4-arch2-1 Electron: 25.8.1
> 
> I think there was problems doing multiple migrations after a long time and it broke. I can't say for sure that it is rotki's fault, perhaps one migration is ended abrupty (it was not responding for a long time and I killed the terminal).

If the version you started from was too old then that's probably it. Migrations from too old versions are not supported. It's probably easier, if you haven't used rotki for so long to just start clean. Remove all data (or better move to some kind of backup) and start again. For support like that please just join ourdiscord as Yabir pointed out above.
